pkg/proxy/handler/detected_labels.go

```diff
@@ -0,0 +1,128 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"sort"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/paulojmdias/lokxy/pkg/o11y/metrics"
+	traces "github.com/paulojmdias/lokxy/pkg/o11y/tracing"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// DetectedLabel represents a single detected label entry from Loki
+type DetectedLabel struct {
+	Label       string `json:"label"`
+	Cardinality int    `json:"cardinality"`
+}
+
+// LokiDetectedLabelsResponse represents the structure of the detected labels response from Loki
+type LokiDetectedLabelsResponse struct {
+	DetectedLabels []DetectedLabel `json:"detectedLabels"`
+}
+
+// HandleLokiDetectedLabels aggregates detected labels from multiple Loki instances
+func HandleLokiDetectedLabels(w http.ResponseWriter, results <-chan *http.Response, logger log.Logger) {
+	ctx, span := traces.CreateSpan(context.Background(), "handle_detected_labels")
```

Comment:
```suggestion
	ctx := r.Context()  
	ctx, span := traces.CreateSpan(context.Background(), "handle_detected_labels")
```
to avoid creating a new context, other than that lgtm

Reply:
I changed a bit to receive the context from the `ProxyHandler`. All the needed changes for all the components will be done on enablement at #99